web/src/views/InvestigationDetail.vue

```diff
@@ -498,17 +505,84 @@ export default defineComponent({
       }));
     }
 
+    async function toggleNonTiledImageOverlay(pin: Pin, dataset: Dataset) {
+      if (!dataset.content) return;
+      if (!featureLayer) return; // sanity check. this should exist
```

Comment:
In both of these cases, if the check represents an impossible condition (placed into an if statement to satisfy the typechecker) then it is probably better to explicitly throw an `Error` object rather than silently `return`ing.

It's also a best practice, stylewise, to place single statement blocks inside curly brackets, to aid in legibility and prevent accidental problems when adding more statements to the body of such a block.

Reply:
see 1fe6e72